bittensor/_subtensor/subtensor_impl.py

```diff
@@ -1667,3 +1711,7 @@ def neuron_for_wallet( self, wallet: 'bittensor.Wallet', block: int = None ) ->
                 neuron object associated with uid or None if it does not exist.
         """
         return self.neuron_for_pubkey ( wallet.hotkey.ss58_address, block = block )
+
+class NeuronPullException(Exception):
+    """Raised when pullin a Neuron from the chain fails."""
+    pass
```

Comment:
Could we use a Python built-in exception here?

Is python idiomatic to use built-in exceptions.

check out this list: https://docs.python.org/3/library/exceptions.html

Maybe **LookupError** ?

```
exception LookupError

    The base class for the exceptions that are raised when a key or index used on a mapping 
    or sequence is invalid: IndexError, KeyError. This can be raised directly by codecs.lookup().
```

- You can check [codecs.lookup() here](https://docs.python.org/3/library/codecs.html#codecs.lookup)

Or maybe:
- [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError)
- [KeyError](https://docs.python.org/3/library/exceptions.html#KeyError)

Reply:
I think it's usually best to create your own exceptions so you can catch them properly.

in this case, the exception inherits from a base exception for fast sync, so I can do `except FastSyncException` but I can also be more specific and except for the exact exception type.

If I made it a built-in exception, then a bug in my code may also raise this built-in and I would unknowingly handle it with the wrong assumption

Reply:
I see your point.

I like this consideration you can find in [PEP recommendations](https://peps.python.org/pep-0008/#programming-recommendations):

> Design exception hierarchies based on the distinctions that code catching the exceptions is likely to need, rather than the locations where the exceptions are raised. Aim to answer the question “What went wrong?” programmatically, rather than only stating that “A problem occurred” (see [PEP 3151](https://peps.python.org/pep-3151) for an example of this lesson being learned for the builtin exception hierarchy)

so, if this case is something that we need to know to differentiate the way we handle the error, it makes sense, but if not...are we using this to trace the error in case of debugging?

Reply:
I believe I'm using it in the former way.

The idea is that if an exception occurs, I know what it's caused by (i.e. what went wrong), so I can handle it correctly
And that way I don't handle an exception raised by something that I'm not aware of (e.g. a `LookupError` that isn't caused by having an neuron pull)

Reply:
fair enough 👍🏽